js/kucoinfutures.js

```diff
@@ -0,0 +1,1308 @@
+                    'tiers': {
+                        'taker': [
+                            [ this.parseNumber ('0'), this.parseNumber ('0.0006') ],
+                            [ this.parseNumber ('50'), this.parseNumber ('0.0006') ],
+                            [ this.parseNumber ('200'), this.parseNumber ('0.0006') ],
+                            [ this.parseNumber ('500'), this.parseNumber ('0.0005') ],
+                            [ this.parseNumber ('1000'), this.parseNumber ('0.0004') ],
+                            [ this.parseNumber ('2000'), this.parseNumber ('0.0004') ],
+                            [ this.parseNumber ('4000'), this.parseNumber ('0.00038') ],
+                            [ this.parseNumber ('8000'), this.parseNumber ('0.00035') ],
+                            [ this.parseNumber ('15000'), this.parseNumber ('0.00032') ],
+                            [ this.parseNumber ('25000'), this.parseNumber ('0.0003') ],
+                            [ this.parseNumber ('40000'), this.parseNumber ('0.0003') ],
+                            [ this.parseNumber ('60000'), this.parseNumber ('0.0003') ],
+                            [ this.parseNumber ('80000'), this.parseNumber ('0.0003') ],
+                        ],
+                        'maker': [
+                            [ this.parseNumber ('0'), this.parseNumber ('0.02') ],
+                            [ this.parseNumber ('50'), this.parseNumber ('0.015') ],
+                            [ this.parseNumber ('200'), this.parseNumber ('0.01') ],
+                            [ this.parseNumber ('500'), this.parseNumber ('0.01') ],
+                            [ this.parseNumber ('1000'), this.parseNumber ('0.01') ],
+                            [ this.parseNumber ('2000'), this.parseNumber ('0') ],
+                            [ this.parseNumber ('4000'), this.parseNumber ('0') ],
+                            [ this.parseNumber ('8000'), this.parseNumber ('0') ],
+                            [ this.parseNumber ('15000'), this.parseNumber ('-0.003') ],
+                            [ this.parseNumber ('25000'), this.parseNumber ('-0.006') ],
+                            [ this.parseNumber ('40000'), this.parseNumber ('-0.009') ],
+                            [ this.parseNumber ('60000'), this.parseNumber ('-0.012') ],
+                            [ this.parseNumber ('80000'), this.parseNumber ('-0.015') ],
+                        ],
+                    },
+                },
+                'funding': {
+                    'tierBased': false,
+                    'percentage': false,
+                    'withdraw': {},
+                    'deposit': {},
+                },
+            },
+            'commonCurrencies': {
+                'HOT': 'HOTNOW',
+                'EDGE': 'DADI', // https://github.com/ccxt/ccxt/issues/5756
+                'WAX': 'WAXP',
+                'TRY': 'Trias',
+                'VAI': 'VAIOT',
+                'XBT': 'BTC',
```

Comment:
I tried swapping the symbol XBT for BTC, but in responses the symbol is still showing up as XBT, like in `fetchBalance`

Reply:
Make sure we use safeCurrencyCode in fetchBalance's response parsing code for that.

Reply:
I do

Reply:
Can you debug and locate the place where it should happen but does not happen? In general, it should work without adding XBT: BTC to commonCurrencies explicitly, cause that is already here: 

- https://github.com/ccxt/ccxt/blob/master/js/base/Exchange.js#L179
- https://github.com/ccxt/ccxt/blob/master/python/ccxt/base/exchange.py#L329

The call to `safeCurrencyCode` should do the work with respect to that mapping: 

- https://github.com/ccxt/ccxt/blob/master/js/base/Exchange.js#L1350

You can add some breakpoints or logging there to track the bug down.

Reply:
viv44kE2fhDFaJsczs4R6bt9ofB8pSHPMod8uRWfuZPH5bC96Ndx2bpfYGYtqcBTj27QCx3QucnxzxKtLJtrecu

Reply:
I think it originates from mixing the kucoin's spot currencies with the futures markets by inheritance.
Just by looking at the code, maybe we should split them into separate exchange-files and don't inherit one from another. Or set `exchange.has['fetchCurrencies']` to `false` on kucoinfutures. Whatever works better.

Reply:
I'm not entirely sure tho, it has to be debugged to find the exact cause first.

Reply:
`exchange.has['fetchCurrencies']` didn't work

> maybe we should split them into separate exchange-files and don't inherit one from another
 
 Then I run into [these issues](https://github.com/ccxt/ccxt/pull/10605#discussion_r750158633)
 
 ---------------
 
 You can just run `kucoinfutures fetchBalance` from the command line to test it to test it
 
 ```
 % kucoinfutures fetchBalance
2021-12-08T03:32:20.583Z
Node.js: v14.17.0
CCXT v1.63.10
kucoinfutures.fetchBalance ()
269 ms
{      info: { code:   "200000",
               data: {    accountEquity:  0.00028298,
                          unrealisedPNL:  0,
                          marginBalance:  0.00028298,
                         positionMargin:  0,
                            orderMargin:  0,
                            frozenFunds:  0,
                       availableBalance:  0.00028298,
                               currency: "XBT"        } },
  timestamp:    undefined,
   datetime:    undefined,
        XBT: { free: 0.00028298, used: 0, total: 0.00028298 },
       free: { XBT: 0.00028298 },
       used: { XBT: 0 },
      total: { XBT: 0.00028298 }                               }
```

Reply:
Ok, no worries, I will debug it on my side to find it and fix it.